goldstone/core/tasks.py

```diff
@@ -82,15 +89,17 @@ def update_persistent_graph():
        - updated from the cloud if they are already in the graph.
 
     """
-    from goldstone.cinder.utils import update_nodes as update_cinder_nodes
-    from goldstone.glance.utils import update_nodes as update_glance_nodes
-    from goldstone.keystone.utils import update_nodes as update_keystone_nodes
-    from goldstone.nova.utils import update_nodes as update_nova_nodes
-
-    update_cinder_nodes()
-    update_glance_nodes()
-    update_keystone_nodes()
-    update_nova_nodes()
+
+    graph_resources = [cinder_res, glance_res, keystone_res, nova_res]
+
+    for obj in graph_resources:
```

Comment:
calling all the updates each time through the loop is probably not what you want